engines/stark/gfx/openglsactor.cpp

```diff
@@ -112,6 +116,34 @@ void OpenGLSActorRenderer::render(const Math::Vector3d &position, float directio
 	}
 
 	_shader->unbind();
+
+	if (castsShadow && StarkSettings->getBoolSetting(Settings::kShadow)) {
+		glEnable(GL_BLEND);
+		glEnable(GL_STENCIL_TEST);
```

Comment:
I'm not under the impression the stencil buffer is actually used.

Reply:
See the `glStencilFunc` and `glStencilOp` calls in `set3DMode`. The first makes it so that every touched pixel ends with a stencil value of 1 (or more if overdraw occurs), while the second discards pixels that have a non-zero stencil value. This prevents the shadows from overlapping and creates a uniform shadow shape.